src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +451,338 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
                 iEndIndex = endIndex
             };
 
-            return expansion.InsertExpansion(textSpan, textSpan, this, LanguageServiceGuid, out ExpansionSession) == VSConstants.S_OK;
+            if (expansion.InsertExpansion(textSpan, textSpan, this, LanguageServiceGuid, out ExpansionSession) == VSConstants.S_OK)
+            {
+                // This expansion is not derived from a symbol
+                _symbols = default;
+                _symbol = null;
+                _arguments = _arguments.Clear();
+                return true;
+            }
+
+            if (!(SubjectBuffer.GetFeatureOnOffOption(CompletionOptions.EnableCallCompletionOnTabTab) ?? false))
```

Comment:
not understanidng this check.  isn't EnableCallCompletiononTabTab just a boolean option?  why do we need the coalesce + `!`?

Reply:
Tri-state pattern is used for options that can be set through an experiment.